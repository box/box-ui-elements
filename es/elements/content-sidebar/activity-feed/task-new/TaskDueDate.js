function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import ActivityDatestamp from '../common/activity-datestamp';
import messages from './messages';
// $FlowFixMe LabelPill is in typescript
import LabelPill from '../../../../components/label-pill';
import { TASK_NEW_NOT_STARTED } from '../../../../constants';
const TaskDueDate = ({
  dueDate,
  status
}) => {
  const isOverdue = dueDate ? status === TASK_NEW_NOT_STARTED && new Date(dueDate) < Date.now() : false;
  const fullDueDate = new Date(dueDate);
  const pillProps = isOverdue ? {
    'data-testid': 'task-overdue-date',
    type: 'error'
  } : {
    type: 'default'
  };
  return /*#__PURE__*/React.createElement("div", {
    "data-testid": "task-due-date"
  }, /*#__PURE__*/React.createElement(LabelPill.Pill, pillProps, /*#__PURE__*/React.createElement(LabelPill.Text, null, /*#__PURE__*/React.createElement(FormattedMessage, _extends({}, messages.taskFeedStatusDue, {
    values: {
      dateTime: /*#__PURE__*/React.createElement(ActivityDatestamp, {
        date: fullDueDate.getTime(),
        uppercase: true
      })
    }
  })))));
};
export default TaskDueDate;
//# sourceMappingURL=TaskDueDate.js.map