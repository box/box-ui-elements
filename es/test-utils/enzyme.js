function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map