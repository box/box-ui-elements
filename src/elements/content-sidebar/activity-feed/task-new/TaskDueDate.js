// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import ActivityDatestamp from '../common/activity-datestamp';
import messages from './messages';
// $FlowFixMe LabelPill is in typescript
import LabelPill from '../../../../components/label-pill';
import { TASK_NEW_NOT_STARTED } from '../../../../constants';

import type { TaskStatus } from '../../../../common/types/tasks';
import type { ISODate } from '../../../../common/types/core';

type Props = {
    dueDate: ISODate,
    status: TaskStatus,
};

const TaskDueDate = ({ dueDate, status }: Props): React.Node => {
    const isOverdue = dueDate ? status === TASK_NEW_NOT_STARTED && new Date(dueDate) < Date.now() : false;
    const fullDueDate = new Date(dueDate);
    const pillProps = isOverdue ? { 'data-testid': 'task-overdue-date', type: 'error' } : { type: 'default' };
    return (
        <div data-testid="task-due-date">
            <LabelPill.Pill {...pillProps}>
                <LabelPill.Text>
                    <FormattedMessage
                        {...messages.taskFeedStatusDue}
                        values={{
                            dateTime: <ActivityDatestamp date={fullDueDate.getTime()} uppercase />,
                        }}
                    />
                </LabelPill.Text>
            </LabelPill.Pill>
        </div>
    );
};

export default TaskDueDate;
