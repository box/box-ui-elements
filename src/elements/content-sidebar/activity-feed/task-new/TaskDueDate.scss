@import '../../../common/variables';

.bcs-TaskDueDate {
    color: $bdl-gray-62;
    font-weight: bold;

    &.bcs-is-taskOverdue {
        color: $bdl-watermelon-red;
    }
}
