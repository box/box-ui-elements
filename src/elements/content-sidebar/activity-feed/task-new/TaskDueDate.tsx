import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import ActivityDatestamp from '../common/activity-datestamp';
import messages from './messages';
import LabelPill from '../../../../components/label-pill';
import { TASK_NEW_NOT_STARTED } from '../../../../constants';

import type { TaskStatus } from '../../../../common/types/tasks';
import type { ISODate } from '../../../../common/types/core';

interface TaskDueDateProps {
    dueDate: ISODate;
    status: TaskStatus;
}

const TaskDueDate = ({ dueDate, status }: TaskDueDateProps): JSX.Element => {
    const isOverdue = dueDate ? status === TASK_NEW_NOT_STARTED && new Date(dueDate) < Date.now() : false;
    const fullDueDate = new Date(dueDate);
    const pillProps = isOverdue ? { 'data-testid': 'task-overdue-date', type: 'error' } : { type: 'default' };
    return (
        <div data-testid="task-due-date">
            <LabelPill.Pill {...pillProps}>
                <LabelPill.Text>
                    <FormattedMessage
                        {...messages.taskFeedStatusDue}
                        values={{
                            dateTime: <ActivityDatestamp date={fullDueDate.getTime()} uppercase />,
                        }}
                    />
                </LabelPill.Text>
            </LabelPill.Pill>
        </div>
    );
};

export default TaskDueDate;
