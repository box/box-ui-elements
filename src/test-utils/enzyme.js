/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
