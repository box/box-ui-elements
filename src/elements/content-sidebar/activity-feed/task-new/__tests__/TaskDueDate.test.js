import * as React from 'react';
import { mount } from 'enzyme';

import TaskDueDate from '../TaskDueDate';

describe('elements/content-sidebar/ActivityFeed/task-new/TaskDueDate', () => {
    test('should render the proper DOM structure', () => {
        const dueDate = mount(<TaskDueDate dueDate={new Date(987654321000)} status="NOT_STARTED" />);

        expect(dueDate).toMatchSnapshot();
    });

    test('should add proper class when past due date', () => {
        const dueDate = shallow(<TaskDueDate dueDate={new Date() - 1000} status="NOT_STARTED" />);

        expect(dueDate.find('[data-testid="task-overdue-date"]')).toHaveLength(1);
    });
});
