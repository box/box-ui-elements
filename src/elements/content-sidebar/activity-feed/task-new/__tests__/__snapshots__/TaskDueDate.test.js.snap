// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/content-sidebar/ActivityFeed/task-new/TaskDueDate should render the proper DOM structure 1`] = `
<TaskDueDate
  dueDate={2001-04-19T04:25:21.000Z}
  status="NOT_STARTED"
>
  <div
    data-testid="task-due-date"
  >
    <LabelPill
      data-testid="task-overdue-date"
      type="error"
    >
      <span
        className="bdl-LabelPill bdl-LabelPill--error bdl-LabelPill--sizeRegular"
        data-testid="task-overdue-date"
      >
        <LabelPillText>
          <span
            className="bdl-LabelPill-textContent"
          >
            <FormattedMessage
              defaultMessage="DUE {dateTime}"
              id="be.contentSidebar.activityFeed.taskNew.taskFeedStatusDue"
              values={
                Object {
                  "dateTime": <ActivityDatestamp
                    date={987654321000}
                    uppercase={true}
                  />,
                }
              }
            >
              <div />
            </FormattedMessage>
          </span>
        </LabelPillText>
      </span>
    </LabelPill>
  </div>
</TaskDueDate>
`;
